scripts/review_comments_retriver.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import os
+import re
+from collections import defaultdict
+from functools import lru_cache
+
+import tenacity
+from libmozdata.phabricator import PhabricatorAPI
+from tqdm import tqdm
+from unidiff import Hunk, PatchedFile, PatchSet, UnidiffParseError
+
+from bugbug import db, phabricator
+from bugbug.tools.code_review import InlineComment
+from bugbug.utils import get_secret
+from bugbug.vectordb import QdrantVectorDB, ReviewCommentsDB
+
+
+class PhabricatorReviewCommentRetriever:
+    NIT_PATTERN = re.compile(r"[^a-zA-Z0-9]nit[\s:,]", re.IGNORECASE)
+
+    def __init__(self) -> None:
+        os.makedirs("patches", exist_ok=True)
+        self.phabricator_api = PhabricatorAPI(
+            get_secret("PHABRICATOR_TOKEN"),
+            get_secret("PHABRICATOR_URL"),
+        )
+
+        db.download(phabricator.REVISIONS_DB)
+
+    @lru_cache(maxsize=1)
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(7),
+        wait=tenacity.wait_exponential(multiplier=1, min=16, max=64),
+        reraise=True,
+    )
+    def load_patch_set(self, diff_id):
+        try:
+            with open(f"patches/{diff_id}.patch", "r") as f:
+                patch = f.read()
+        except FileNotFoundError:
+            with open(f"patches/{diff_id}.patch", "w") as f:
+                patch = self.phabricator_api.load_raw_diff(diff_id)
+                f.write(patch)
+
+        return PatchSet.from_string(patch)
+
+    def _get_revisions_count(self):
+        return sum(1 for _ in phabricator.get_revisions())
+
+    def retrieve_diff_comments(self):
+        for revision in tqdm(
+            phabricator.get_revisions(), total=self._get_revisions_count()
+        ):
+            diff_comments: dict[int, list[InlineComment]] = defaultdict(list)
+
+            for transaction in revision["transactions"]:
+                if transaction["type"] != "inline":
+                    continue
+
+                # Ignore replies
+                if transaction["fields"]["replyToCommentPHID"] is not None:
+                    continue
+
+                # Ignore bot comments
+                if transaction["authorPHID"] == "PHID-USER-cje4weq32o3xyuegalpj":
+                    continue
+
+                if len(transaction["comments"]) != 1:
+                    print(transaction)
+                    # raise Exception("Unexpected, need to look into it")
+                    continue
```

Comment:
Could you file a follow-up for this?

Reply:
I filed https://github.com/mozilla/bugbug/issues/4218 to follow-up on this.